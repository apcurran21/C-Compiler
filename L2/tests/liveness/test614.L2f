(@compute_many_tns
	1
	%nums <- rdi
	%decoded_nums <- %nums
	%decoded_nums >>= 1
	%nth_tn <- 3
	%nth_tn >>= 1
	:call_label_ciao_2
	%cond <- %decoded_nums < %nth_tn
	cjump %cond = 1 :call_label_ciao_5
	:call_label_ciao_3
	%call_nth_tn <- %nth_tn
	%call_nth_tn <<= 1
	%call_nth_tn += 1
	rdi <- %call_nth_tn
	mem rsp -8 <- :call_label_ciao_6
	call @compute_triangular_num 1
	:call_label_ciao_6
	goto :call_label_ciao_4
	:call_label_ciao_4
	%nth_tn += 1
	goto :call_label_ciao_2
	:call_label_ciao_5
	return
)
