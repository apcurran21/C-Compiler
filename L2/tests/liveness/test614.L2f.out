(
(in
(r12 r13 r14 r15 rax rbp rbx rdi)
(%nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%cond %decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx rdi)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx rdi)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%cond %decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%call_nth_tn %decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx rdi)
(%decoded_nums %nth_tn r12 r13 r14 r15 rbp rbx rdi)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(%decoded_nums %nth_tn r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
